File: src/main/java/com/bermudalocket/terracotta/Terracotta.java
package com.bermudalocket.terracotta;

import com.bermudalocket.terracotta.command.ButcherCommand;
import com.bermudalocket.terracotta.command.EntityCountCommand;
import org.bukkit.Bukkit;
import org.bukkit.plugin.java.JavaPlugin;

public class Terracotta extends JavaPlugin {

    @Override
    public void onEnable() {
        ButcherCommand.register();
        EntityCountCommand.register();
    }

    public static int getViewDistance() {
        return Bukkit.getServer().getViewDistance() * 16;
    }

}

File: src/main/java/com/bermudalocket/terracotta/Terracotta.kt
package com.bermudalocket.terracotta

import br.com.devsrsouza.kotlinbukkitapi.architecture.KotlinPlugin
import br.com.devsrsouza.kotlinbukkitapi.extensions.event.event
import br.com.devsrsouza.kotlinbukkitapi.extensions.event.events
import com.bermudalocket.terracotta.command.ButcherCommand
import com.bermudalocket.terracotta.command.EntityCountCommand
import com.destroystokyo.paper.event.block.TNTPrimeEvent
import com.destroystokyo.paper.event.block.TNTPrimeEvent.PrimeReason.EXPLOSION
import org.bukkit.Bukkit
import org.bukkit.event.Listener

class Terracotta: KotlinPlugin(), Listener {

    companion object {
        fun getViewDistance(): Double {
            return Bukkit.getServer().viewDistance * 16.0
        }
    }

    override fun onPluginEnable() {
        this.getCommand("terrabutcher")?.setExecutor(ButcherCommand)
        this.getCommand("terracounter")?.setExecutor(EntityCountCommand)
        this.events {
            // prevent TNT chain reactions
            event<TNTPrimeEvent> {
                if (reason == EXPLOSION) isCancelled = true
            }
        }
    }

}
